grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java

```diff
@@ -319,11 +338,26 @@ private void closeListener(Status newStatus) {
                 // The response is streamed so we don't have anything to set as the RpcResponse, so we set it
                 // arbitrarily so it can at least be counted.
                 ctx.logBuilder().responseContent(RpcResponse.of("success"), null);
-                listener.onComplete();
+                try (SafeCloseable ignored = RequestContext.push(ctx)) {
+                    listener.onComplete();
+                } catch (Throwable t) {
+                    // This should not be possible with normal generated stubs which do not implement
+                    // onComplete, but is conceivable for a completely manually constructed stub.
+                    logger.warn("Error in grpc onComplete handler.", t);
```

Comment:
Heh - yeah gRPC seems correct. Updated this string, will send another PR to update any other occurrences in strings or comments.